### PR TITLE
Create ResponseObjectFieldMixin and beginning work on applying it

### LINKED ISSUE #103: Response object fields should match glossary
Our response object fields are out of date. The new design has each object share all fields except applications which have two extra fields.

To keep our code more DRY and set the stage for an eventual data model change, I think we should use a mixin to define shared fields and apply them to multiple tables.

### PR DESCRIPTION
**Pull request by rmarow**
resolves #103

### CODE REVIEW

File: usaon_vta_survey/models/tables.py [hunk 1]
```diff
@@ -64,6 +64,26 @@ def __io__(cls) -> IORelationship:
         return io
 
 
+class ResponseObjectFieldMixin:
```
Comment (rmarow) [inline usaon_vta_survey/models/tables.py:67]: comments contain questions for hazel/matt

File: usaon_vta_survey/templates/macros/forms/misc.j2 [hunk 2]
```diff
@@ -1,38 +1,38 @@
 {% macro observing_system_fields(form) -%}
```
Comment (rmarow) [inline usaon_vta_survey/templates/macros/forms/misc.j2:1]: likely can rename this something like `response_object_fields` and use for all besides `Application`

File: usaon_vta_survey/models/tables.py [hunk 3]
```diff
@@ -64,6 +64,26 @@ def __io__(cls) -> IORelationship:
...
+    """Provide shared fields between all relationshio objects to reduce repition."""
+
+    short_name = Column(String(256), nullable=False)
+    full_name = Column(String(256), nullable=False)
+    organization = Column(String(256), nullable=False)
+    # should this be funding agency?
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:73]: @hazelshapiro

File: usaon_vta_survey/templates/macros/forms/misc.j2 [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
+  {{form.short_name.label}} {{form.short_name(size=50)}}
+  {% if form.short_name.errors %}
     <ul class="errors">
-    {% for error in form.name.errors %}
+    {% for error in form.short_name.errors %}
       <li>{{ error }}</li>
     {% endfor %}
     </ul>
   {% endif %}
   <br />
```
Comment (rmarow) [inline usaon_vta_survey/templates/macros/forms/misc.j2:10]: need to add in rest of missing fields.

File: usaon_vta_survey/models/tables.py [hunk 5]
```diff
@@ -64,6 +64,26 @@ def __io__(cls) -> IORelationship:
...
+    funder = Column(String(256), nullable=False)
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:74]: @hazelshapiro

File: usaon_vta_survey/models/tables.py [hunk 6]
```diff
@@ -64,6 +64,26 @@ def __io__(cls) -> IORelationship:
...
+    funding_country = Column(String(256), nullable=False)
+    # clairify this
+    website_url = Column(String(256), nullable=True)
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:77]: @rmarow can you clarify your clarify comment? :) What do you need to know?

File: usaon_vta_survey/models/tables.py [hunk 7]
```diff
@@ -64,6 +64,26 @@ def __io__(cls) -> IORelationship:
...
+    description = Column(String(512), nullable=True)
+    contact_name = Column(String(256), nullable=False)
+    contact_title = Column(String(256), nullable=True)
+    contact_email = Column(String(256), nullable=False)
+    # how do i make it a minimum of 3 tags?
+    tags = Column(String, nullable=False)
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:83]: @rmarow I'm not sure I understand this. I looked in the glossary and didn't see anything about a minimum number of tags. I think this should be an [Array column](https://www.postgresql.org/docs/current/arrays.html) ([sqlalchemy](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.ARRAY))!

Comment (hazelshapiro) [reply usaon_vta_survey/models/tables.py:83]: If this is hard to code, we can just add explanatory text that says "recommend three or more tags." Sandy had suggested this last week, and I added a note in the front-end document, which @rmarow noted. I think the explanatory text is a good way to go. If we do some testing and folks aren't responding to that request, we can adjust.

File: usaon_vta_survey/models/tables.py (hunk 5 above)
Comment (hazelshapiro) [reply usaon_vta_survey/models/tables.py:74]: The "funder" would refer to a funding agency (or another group if it's not federally funded), while "organization" refers to the group that manages and runs the system. In some cases, they will be the same.

File: usaon_vta_survey/forms.py
Comment (rmarow) [inline usaon_vta_survey/forms.py:1]: @MattF-NSIDC is there a way to reduce the repetitiveness here>

File: usaon_vta_survey/forms.py [hunk 8]
```diff
@@ -62,24 +62,56 @@ def conv_String(self, field_args, **extra):
...
+            'version',
         ],
     ),
     # TODO: Restrict "rating" values to correct range
     ResponseDataProduct: model_form(
         ResponseDataProduct,
-        only=['name', 'performance_rating'],
+        only=[
+            'short_name',
+            'full_name',
```
Comment (rmarow) [inline usaon_vta_survey/forms.py:84]: @MattF-NSIDC is there a way to reduce the repetitiveness here?

File: usaon_vta_survey/models/tables.py (hunk 7 above)
Comment (MattF-NSIDC) [reply usaon_vta_survey/models/tables.py:80]: I agree, let's go with explanatory text and treat the actual validation as icing for later :)

File: usaon_vta_survey/forms.py [hunk 9]
```diff
@@ -29,6 +29,21 @@
     User,
 )
 
+response_object_fields = [
```
[the remark this answers is not in the document]
Comment (MattF-NSIDC) [reply usaon_vta_survey/forms.py:32]: Also, is there a way we can generate this list from the mixin class? There are _many_ ways to get the "fields" out of a class, but they're all pretty messy. We could make `ResponseObjectFieldMixin` a pydantic class, and then I think we have more intuitive access to its fields. Or we could use one of these methods:

https://stackoverflow.com/questions/9058305/getting-attributes-of-a-class

I don't feel we need to address this in this PR, except to leave a comment for future us that some DRYing needs to be done.

File: usaon_vta_survey/models/tables.py [hunk 10]
```diff
@@ -213,14 +226,6 @@ class ResponseObservingSystem(BaseModel, IORelationshipMixin):
         'polymorphic_on': type,
     }
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:227]: `id` and `response_id` are also common fields. Can those go in the mixin as well?

**Review by MattF-NSIDC — APPROVED** [review]
LGTM, just a few comments to address that I think shouldn't require a second review :) Let me know if you want me to take a look again before the merge though! Happy to do it :)

File: usaon_vta_survey/models/tables.py (hunk 10 above)
Comment (rmarow) [reply usaon_vta_survey/models/tables.py:227]: Sqlalchemy did not like when i tried to do this.

**Pull request merged**